plugins/tmux/tmux.plugin.sh

```diff
@@ -0,0 +1,3 @@
+# @chopnico 2021
+
+[ -z "$TMUX"  ] && { tmux -2 -u attach || exec tmux -2 -u new-session && exit;}
```

Comment:
May I confirm the intent of this line? In shell syntax, `||` and `&&` have effectively the same precedence and left associativity (unlike typical languages like C/C++, JavaScript, etc.), so the above code is actually interpreted as

```bash
[ -z "$TMUX"  ] && { { tmux -2 -u attach || exec tmux -2 -u new-session; } && exit; }
```

but then the question is why don't you simply write

```bash
[ -z "$TMUX"  ] && { tmux -2 -u attach || exec tmux -2 -u new-session; } && exit
```

Possibly you have intended the following code?

```bash
[ -z "$TMUX"  ] && { tmux -2 -u attach || { exec tmux -2 -u new-session && exit; } }
```

Reply:
No, you're correct.

```bash 
[ -z "$TMUX"  ] && { tmux -2 -u attach || exec tmux -2 -u new-session; } && exit
```

I wanted the bash session to completely end once all tmux sessions were terminated including closing out my terminal.

The last example brought up an interesting question, though, surrounding detaching sessions. I could add a variable such as "OMB_TMUX_BEHAVIOR" that would allow you to configure who TMUX closes.

The suggested format, and my current preferred intent, also closes the terminal if you detach from the session. If I added something like this:

```bash
[ -z "$OMB_TMUX_BEHAVIOR" ] && export OMB_TMUX_BEHAVIOR="detach"

tmux_exit(){
	[ -z "$TMUX"  ] && { tmux -2 -u attach || exec tmux -2 -u new-session; } && exit
}
tmux_detach(){
	[ -z "$TMUX"  ] && { tmux -2 -u attach || tmux -2 -u new-session; }
}

case "$OMB_TMUX_BEHAVIOR" in
	"exit")
		tmux_exit
		;;
	"detach")
		tmux_dettach
		;;
	*)
		tmux_dettach
		;;
esac
```

We could allow for that behavior to change.

Reply:
Thank you for your reply! I now see your original intent.

----

> ```bash
> tmux_exit(){
> 	[ -z "$TMUX"  ] && { tmux -2 -u attach || exec tmux -2 -u new-session; } && exit
> }
> tmux_detach(){
> 	[ -z "$TMUX"  ] && { tmux -2 -u attach || tmux -2 -u new-session; }
> }
> ```

Are those functions also intended to be directly called by users?

----

> "OMB_TMUX_BEHAVIOR"

- Currently, there is no explicit naming convention for the config variables of `oh-my-bash`, but it seems like `OSH_` is the prefix used by `oh-my-bash`. The main script `oh-my-bash.sh` and a few themes use these variables. There are no existing instances of variables named `OMB_*`. The theme variable names seem to have the form `OSH_THEME_${ThemeName^^}_${ConfigName^^}`.
- `BEHAVIOR` seems too general.
- Actually, the plugin name `tmux` feels also too general. This plugin specifically attaches to a `tmux` session on the startup of the shell.

I'd suggest changing the plugin name from `tmux` to `tmux-attach`, `tmux-autoattach`, or else. For the configuration name, I'd suggest something like `OSH_PLUGIN_TMUX_ATTACH_BEHAVIOR`. Maybe we need to shift from `OSH_*` to `OMB_*`, but I feel that that should be discussed in a separate PR or issue.

----

By the way, there seems like [an option `-A` for `new-session`](https://stackoverflow.com/a/49134974/4908404) in tmux 1.8 (2013/03/29), so that we may just write `tmux -2u new -A` instead of `tmux -2 -u attach || tmux -2 -u new-session`. But, of course, that doesn't work for older `tmux`.

Reply:
Awesome.

I think I'll remove the functions. I forget we're sourcing these so maybe let's not unnecessarily clutter the session?

I like the OSH_PLUGIN_TMUX_ATTACH_BEHAVIOR, so I'll continue with that naming convention. I'll also update the name to tmux-autoattach as I think that best represents the action here.

I think I'm OK with moving to `tmux -2u new -A`. I'll make a note that it only supports 1.8 and up.

Reply:
> I think I'll remove the functions. I forget we're sourcing these so maybe let's not unnecessarily clutter the session?

Or, maybe we can use `_osh_plugin_tmux_autoattach_{exit,detach}` for the function names.

> I like the OSH_PLUGIN_TMUX_ATTACH_BEHAVIOR, so I'll continue with that naming convention.

OK! I think I'll later open a discussion on the naming convention.

> I'll also update the name to tmux-autoattach as I think that best represents the action here.

In that case, I'd think the appropriate name should be `OSH_PLUGIN_TMUX_AUTOATTACH_BEHAVIOR`. I'd like to confine the related variables in a plugin-specific namespace (`TMUX_AUTOATTACH` in this case).

> I think I'm OK with moving to tmux -2u new -A. I'll make a note that it only supports 1.8 and up.

Oh, sorry. I think I need to make it clear that I didn't intend to request shifting to `tmux -2u new -A`, but I just found it by searching. It's completely up to you; you can choose either way as you like.

Reply:
Excellent.

I think ` _osh_plugin_tmux_autoattach_{exit,detach}` would work just wonderfully. 

> In that case, I'd think the appropriate name should be OSH_PLUGIN_TMUX_AUTOATTACH_BEHAVIOR. I'd like to confine the related variables in a plugin-specific namespace (TMUX_AUTOATTACH in this case).

Fantastic. 

> Oh, sorry. I think I need to make it clear that I didn't intend to request shifting to tmux -2u new -A, but I just found it by searching. It's completely up to you; you can choose either way as you like.

I like the `tmux -2u new -A`. That was a wonderful find and it cleans it up a bit.

Reply:
> I think ` _osh_plugin_tmux_autoattach_{exit,detach}` would work just wonderfully.

OK! Thank you! I have asked the original maintainer @nntoan about the naming convention at https://github.com/ohmybash/oh-my-bash/discussions/268#discussioncomment-1866155. (I have just become a new maintainer yesterday as @nntoan seems to be busy). I'm willing to shift to `OMB_*` finally if there are no problems, so maybe these functions and variables will be renamed again later.

> I like the tmux -2u new -A. That was a wonderful find and it cleans it up a bit.

Great!

Reply:
Awesome. Welcome aboard and thank you for all your help!

Reply:
Thank you!